### ISSUE TITLE
/dev/shm在运行过程中不断增大

### ISSUE THREAD

**Issue by zh-hike** [PaddlePaddle/Paddle#49840]
### 请提出你的问题 Please ask your question

DataLoader:
      num_workers: 12
      use_shared_memory: true
      prefetch_factor: 32
      batch_sampler:
        name: DistributedBatchSampler
        batch_size: 32
        shuffle: false
        drop_last: false
<img width="652" alt="image" src="https://user-images.githubusercontent.com/57249890/212536884-14b031eb-e4e8-479b-9454-99db98456efb.png">
这个 /dev/shm越来越大，请问有什么解决方案么，或者这个 /dev/shm越来越大是什么原因造成的呢, 训练一个epoch的代码如下

```
def train_epoch_common(engine, epoch_id, iter_start=1):
    start_time = time.time()
    engine.time_info['batch_cost'].reset()
    engine.time_info['read_cost'].reset()
    step_per_epoch = engine.cfg['Global']['step_per_epoch']
    engine.train_loss_info.reset()
    train_dataloader_iter = iter(engine.train_dl)
    for iter_id in range(iter_start, step_per_epoch+1):
        try:
            batch = next(train_dataloader_iter)
        except:
            train_dataloader_iter = iter(engine.train_dl)
            batch = next(train_dataloader_iter)

        inputs, targets = batch
        engine.time_info['read_cost'].update(time.time() - start_time)

        if engine.amp:
            with paddle.amp.auto_cast(level=engine.amp_level):
                pred = engine.model(inputs)
                loss_dict = engine.train_loss_func(pred, targets)
        else:
            pred = engine.model(inputs)
            loss_dict = engine.train_loss_func(pred, targets)

        loss = loss_dict['loss']
        engine.opt.clear_grad()
        if engine.amp:
            scaled = engine.grad_scaler.scale(loss)
            scaled.backward()
            engine.grad_scaler.minimize(engine.opt, scaled)
        else:
            loss.backward()
            engine.opt.step()

        engine.train_loss_info.update(loss_dict)
        if engine.schedule_update_by == 'step':
            engine.lr.step()

        if iter_id % engine.cfg['Global']['print_batch_step'] == 0:
            log_train_info(engine, epoch_id, iter_id)
            engine.info_reset()

        engine.time_info['batch_cost'].update(time.time() - start_time)
        start_time = time.time()

    del train_dataloader_iter

    if engine.schedule_update_by == 'epoch':
        engine.lr.step()
```

**Comment by ZzSean** [comment]
你好，这个应该是跟dataloader开启use_shared_memory有关，可以尝试调小num_workers或者设置use_shared_memory为False

**Comment by zh-hike** [comment]
> 

你好，当我尝试use_shared_memory = False时，出现
<img width="991" alt="image" src="https://user-images.githubusercontent.com/57249890/212542280-e0d0013b-6510-4e43-83e4-d3856bd76e98.png">
如果将num_workers调小应该也不行，如果设置为0的话跑的实在是太慢了

**Comment by zh-hike** [comment]
这个是目前的配置，使用双卡 2090
<img width="327" alt="image" src="https://user-images.githubusercontent.com/57249890/212542536-aaab2cc2-2776-4b1d-b830-be426767bfbc.png">

这个是报错的日志：
<img width="947" alt="image" src="https://user-images.githubusercontent.com/57249890/212542516-519fffc1-b5bf-4467-bb2a-34cce37d80f7.png">

**Comment by zh-hike** [comment]
目前问题基本解决，

**Comment by idontkonwher** [comment]
请问是怎么解决的？
